bluemira/base/tools.py

```diff
@@ -93,6 +124,286 @@ def create_compound_from_component(comp: Component) -> BluemiraCompound:
+            mat_to_comps_map[mat_name] = []
+        mat_to_comps_map[mat_name].append(phy_comp)
+    return mat_to_comps_map
+
+
+def _build_compounds_from_map(
+    mat_to_comps_map: dict[str, list[PhysicalComponent]],
+    manager_name: str,
+) -> list[PhysicalComponent]:
+    """
+    Build the compounds from the material to components map.
+
+    Returns
+    -------
+    :
+        A list of compounds
+    """
+    return [
+        Component(
+            f"{manager_name}_{mat_name}" if mat_name else manager_name,
+            children=comps,
+        )
+        # compound_from_components(
+        #     comps,
+        #     f"{manager_name}_{mat_name}" if mat_name else manager_name,
+        # )
+        if len(comps) > 1
+        else PhysicalComponent(
+            name=f"{manager_name}_{comps[0].name}",
+            shape=comps[0].shape,
+            material=comps[0].material,
+        )
+        for mat_name, comps in mat_to_comps_map.items()
+    ]
+
+
+def build_comp_manager_save_xyz_cad_tree(
+    comp_manager: ComponentManager,
+    component_filter: Callable[[ComponentT], bool] | None,
+    n_sectors: int,
+    sector_degrees: int,
+) -> Component:
+    """
+    Build the CAD of the component manager's components
+    and save the CAD to a file.
+
+    Parameters
+    ----------
+    comp_manager:
+        Component manager
+    component_filter:
+        Filter to apply to the components
+
+    Returns
+    -------
+    :
+        The constructed component manager component for CAD saving
+
+    Raises
+    ------
+    ValueError
+        If no components were constructed
+    """
+    constructed_phy_comps, manager_name = _construct_comp_manager_physical_comps(
+        comp_manager,
+        component_filter,
+        n_sectors,
+        sector_degrees,
+    )
+
+    if not constructed_phy_comps:
+        raise ValueError(f"No components were constructed for {manager_name}")
+
+    mat_to_comps_map = _group_physical_components_by_material(constructed_phy_comps)
+
+    return_comp = Component(name=manager_name)
+    return_comp.children = _build_compounds_from_map(
+        mat_to_comps_map,
+        manager_name,
+    )
+
+    return return_comp
+
+
+def build_comp_manager_show_cad_tree(
+    comp_manager: ComponentManager,
+    dim: str,
+    component_filter: Callable[[ComponentT], bool] | None,
+    n_sectors: int,
+    sector_degrees: int,
+) -> Component:
+    """
+    Build the CAD of the component manager's components
+    and save the CAD to a file.
+
+    Parameters
+    ----------
+    comp_manager:
+        Component manager
+    dim:
+        Dimension to build the CAD in
+    component_filter:
+        Filter to apply to the components
+    """
+    manager_comp = comp_manager.component()
+    filtered_comp = copy_and_filter_component(
+        manager_comp,
+        dim,
+        component_filter,
+    )
+    if dim == "xyz":
+        filtered_comp = circular_pattern_xyz_components(
```

Comment:
so this will change to accomodate construction types?

Reply:
Yes once I/we figure out how to do colours properly